platforms/documentation/docs/src/docs/userguide/authoring-builds/tasks/lazy_configuration.adoc

```diff
@@ -321,6 +321,30 @@ $ gradle show
 include::{snippetsPath}/providers/propertyConvention/tests/propertyConvention.out[]
 ----
 
+=== Where to apply conventions from?
+
+In general, you want to apply convention to a property at configuration time, before tasks execute, so task actions are not appropriate.
+Still, there are multiple options for appropriate locations for setting a convention on a property.
+
+==== From a plugin's `apply()` method
+
+Plugin authors may configure a convention on a lazy property from a plugin's `apply()` method, while performing preliminary configuration of the property-owning task.
+
+==== From a build script
+
+Build engineers may configure a convention on a lazy property from shared build logic that is configuring tasks in a standard way for the entire build.
```

Comment:
💅 It might be worth it to call out the use cases explicitly:
* when I'm *defining* a property in a plugin/task: I use apply/constructor/initialization
* when I'm configuring a property of a third-party plugin I've applied: I use my convention plugin or `set` in the Project.